components/scream/src/physics/share/physics_universal_impl.hpp

```diff
@@ -106,7 +106,31 @@ Functions<S,D>::get_dz(const Spack& zi_top, const Spack& zi_bot, const Smask& ra
   return result;
 }
 //-----------------------------------------------------------------------------------------------//
+// Compute dry static energy (DSE).
+// The result unit is in J/kg
+// The inputs are
+//   T_mid is the atmospheric temperature. Units in K.
+//   z_mid is the geopotential height above surface at midpoints. Units in m.
+//   surf_geopotential is the surface geopotential height. Units in m.
+template <typename S, typename D>
+KOKKOS_FUNCTION
+typename Functions<S,D>::Spack
+Functions<S,D>::get_dse(const Spack& T_mid, const Spack& z_mid, const Real surf_geopotential, const Smask& range_mask)
+{
+  static constexpr Scalar cp  = C::CP;
+  static constexpr Scalar ggr = C::gravit;
+
+  const Spack dse(range_mask, cp*T_mid + ggr*z_mid + surf_geopotential);
```

Comment:
@PeterCaldwell On the Field manager variable page I see two versions of DSE:
```
dse = cp*T_mid + g*(z_mid+surf_geopotential) (in the table)
```
and, from a Mark Taylor comment on that page, 
```
state%s(:ncol,k) = state%t(:ncol,k  )*cpairv_loc(:ncol,k,state%lchnk)&
                       + gravit*state%zm(:ncol,k) + state%phis(:ncol) 
```
Not sure the source of the latter..

Do you know which is correct? `dse = cp*T_mid + g*(z_mid + surf_geopotential) ` or
`dse = cp*T_mid + g*z_mid + surf_geopotential`

Reply:
@tcclevenger - so the question is whether to multiply grav by phis or not? The answer to this depends on whether phis is geopotential HEIGHT (which I was assuming and should have said in my impl) or actual geopotential (which it appears Mark was assuming). To answer this, we need to check how phis is used elsewhere in the code.

Reply:
check shco_update_host_dse_impl.hpp. It says `host_dse(k) = cp*temp+ggr*zt_grid(k)+phis` so Mark's version is right.

Reply:
Ok, awesome. Thanks!